notebooks/203-meter-reader/203-meter-reader.ipynb

```diff
@@ -0,0 +1,731 @@
+    "    \"\"\"\n",
+    "    match the location of pointer and scales\n",
+    "\n",
+    "    :param：\n",
+    "        scale_locations (list[list])：location of each scale\n",
+    "        pointer_locations (list[list])：location of pointer\n",
+    "\n",
+    "    :return：\n",
+    "        pointed_scales (list[dict])： a list of dict with:\n",
+    "                                     'num_scales': total number of scales\n",
+    "                                     'pointed_scale': predicted number of scales\n",
+    "            \n",
+    "    \"\"\"\n",
+    "    pointed_scales = list()\n",
+    "    for scale_location, pointer_location in zip(scale_locations,\n",
+    "                                                pointer_locations):\n",
+    "        num_scales = len(scale_location)\n",
+    "        pointed_scale = -1\n",
+    "        if num_scales > 0:\n",
+    "            for i in range(num_scales - 1):\n",
+    "                if scale_location[\n",
+    "                        i] <= pointer_location and pointer_location < scale_location[\n",
+    "                            i + 1]:\n",
+    "                    pointed_scale = i + (\n",
+    "                        pointer_location - scale_location[i]\n",
+    "                    ) / (scale_location[i + 1] - scale_location[i] + 1e-05\n",
+    "                            ) + 1\n",
+    "        result = {'num_scales': num_scales, 'pointed_scale': pointed_scale}\n",
+    "        pointed_scales.append(result)\n",
+    "    return pointed_scales\n",
+    "\n",
+    "\n",
+    "def calculate_reading(pointed_scales):\n",
+    "    \"\"\"\n",
+    "    calculate the value of meter according the type of meter\n",
+    "\n",
+    "    :param：\n",
+    "        pointed_scales (list[list])：predicted number of scales\n",
+    "\n",
+    "    :return：\n",
+    "        readings (list[float])： the list of values read from meter\n",
+    "            \n",
+    "    \"\"\"\n",
+    "    readings = list()\n",
+    "    batch_size = len(pointed_scales)\n",
+    "    for i in range(batch_size):\n",
+    "        pointed_scale = pointed_scales[i]\n",
+    "        # find out the type of meter according the total number of scales\n",
+    "        if pointed_scale['num_scales'] > TYPE_THRESHOLD:\n",
+    "            reading = pointed_scale['pointed_scale'] * METER_CONFIG[0][\n",
+    "                'scale_interval_value']\n",
+    "        else:\n",
+    "            reading = pointed_scale['pointed_scale'] * METER_CONFIG[1][\n",
+    "                'scale_interval_value']\n",
+    "        readings.append(reading)\n",
+    "    return readings"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Main Function\n",
+    "Run inference and calculate the meter readings"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def meter_reader(\n",
```

Comment:
This function is long. It would be better to split it into a few cells with code snippets and run them one by one showing results directly under the cell instead of all at the end

Reply:
Done